src/actions/graph.js

```diff
@@ -39,21 +92,42 @@ const chooseLayout = (instance, state) =>
 // Prepare new layout worker
 const layoutWorker = preventWorkerQueues(worker, chooseLayout);
 
+/**
+ * Formula to determine if the pipeline is chonky
+ */
+const isChonky = (chonkyNodeAmount, nodesNo, edgesNo) => {
+  return nodesNo + 1.5 * edgesNo > chonkyNodeAmount ? true : false;
+};
+
 /**
  * Async action to calculate graph layout in a web worker
  * whiled displaying a loading spinner
  * @param {Object} graphState A subset of main state
  * @return {function} A promise that resolves when the calcuation is done
  */
-export function calculateGraph(graphState) {
+export function calculateGraph(graphState, customChonkyLimit) {
```

Comment:
the word custom is used here mainly to indicate this value being a 'custom defined' value to allow the function to accept a figure that would overwrite the limit defined in the config. The main use case for this is for the test, where we would hope to avoid creating a large pipeline just to test this warning, whereby I can pass in a custom limit ( which would be a much smaller amount to fit in with the animals testing dataset) to allow this to be tested without impacting testing performance.